tests/drivers/uart/uart_async_api/src/test_uart.h

```diff
@@ -23,72 +23,13 @@
  * get selected over any other SoC specific uart
  */
 #if defined(CONFIG_UART_RTT)
-#define UART_DEVICE_DEV DT_NODELABEL(rtt0)
-#elif defined(CONFIG_BOARD_NRF52840DK_NRF52840)
-#define UART_DEVICE_DEV DT_NODELABEL(uart0)
-#elif defined(CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP)
-#define UART_DEVICE_DEV DT_NODELABEL(uart1)
-#elif defined(CONFIG_BOARD_NRF9160DK_NRF9160)
-#define UART_DEVICE_DEV DT_NODELABEL(uart1)
-#elif defined(CONFIG_BOARD_ATSAMD21_XPRO)
-#define UART_DEVICE_DEV DT_NODELABEL(sercom1)
-#elif defined(CONFIG_BOARD_ATSAMR21_XPRO)
-#define UART_DEVICE_DEV DT_NODELABEL(sercom3)
-#elif defined(CONFIG_BOARD_ATSAML21_XPRO)
-#define UART_DEVICE_NAME DT_NODELABEL(sercom1)
-#elif defined(CONFIG_BOARD_ATSAMR34_XPRO)
-#define UART_DEVICE_NAME DT_NODELABEL(sercom2)
-#elif defined(CONFIG_BOARD_ATSAME54_XPRO)
-#define UART_DEVICE_DEV DT_NODELABEL(sercom1)
-#elif defined(CONFIG_BOARD_NUCLEO_F103RB) || \
-	defined(CONFIG_BOARD_NUCLEO_G071RB) || \
-	defined(CONFIG_BOARD_NUCLEO_G474RE) || \
-	defined(CONFIG_BOARD_NUCLEO_WL55JC)
-#define UART_DEVICE_DEV DT_NODELABEL(usart1)
-#elif defined(CONFIG_BOARD_NUCLEO_H723ZG) || \
-	defined(CONFIG_BOARD_NUCLEO_H743ZI) || \
-	defined(CONFIG_BOARD_STM32F3_DISCO) || \
-	defined(CONFIG_BOARD_NUCLEO_U575ZI_Q)
-#define UART_DEVICE_DEV DT_NODELABEL(usart2)
-#elif defined(CONFIG_BOARD_NUCLEO_L4R5ZI) || \
-	defined(CONFIG_BOARD_NUCLEO_L152RE) || \
-	defined(CONFIG_BOARD_STM32L562E_DK) || \
-	defined(CONFIG_BOARD_NUCLEO_L552ZE_Q) || \
-	defined(CONFIG_BOARD_B_U585I_IOT02A)
-#define UART_DEVICE_DEV DT_NODELABEL(usart3)
-#elif defined(CONFIG_BOARD_DISCO_L475_IOT1)
-#define UART_DEVICE_DEV DT_NODELABEL(uart4)
-#elif defined(CONFIG_BOARD_NUCLEO_F429ZI) || \
-	defined(CONFIG_BOARD_NUCLEO_F207ZG) || \
-	defined(CONFIG_BOARD_NUCLEO_F767ZI) || \
-	defined(CONFIG_BOARD_NUCLEO_F746ZG)
-#define UART_DEVICE_DEV DT_NODELABEL(usart6)
-#elif defined(CONFIG_BOARD_FRDM_K82F)
-#define UART_DEVICE_DEV DT_NODELABEL(lpuart0)
-#elif defined(CONFIG_BOARD_NUCLEO_WB55RG)
-#define UART_DEVICE_DEV DT_NODELABEL(lpuart1)
-#elif defined(CONFIG_BOARD_MIMXRT1020_EVK) || \
-	defined(CONFIG_BOARD_MIMXRT1024_EVK) || \
-	defined(CONFIG_BOARD_MIMXRT1160_EVK_CM4) || \
-	defined(CONFIG_BOARD_MIMXRT1170_EVK_CM4) || \
-	defined(CONFIG_BOARD_MIMXRT1160_EVK_CM7) || \
-	defined(CONFIG_BOARD_MIMXRT1170_EVK_CM7) || \
-	defined(CONFIG_BOARD_TWR_KE18F)
-#define UART_DEVICE_DEV DT_NODELABEL(lpuart2)
-#elif defined(CONFIG_BOARD_MIMXRT1050_EVK) || \
-	defined(CONFIG_BOARD_MIMXRT1060_EVK) || \
-	defined(CONFIG_BOARD_MIMXRT1060_EVKB) || \
-	defined(CONFIG_BOARD_MIMXRT1064_EVK)
-#define UART_DEVICE_DEV DT_NODELABEL(lpuart3)
-#elif defined(CONFIG_BOARD_MIMXRT1010_EVK) || \
-	defined(CONFIG_BOARD_MIMXRT1015_EVK)
-#define UART_DEVICE_DEV DT_NODELABEL(lpuart4)
+#define UART_NODE DT_NODELABEL(rtt0)
+#elif DT_NODE_EXISTS(DT_NODELABEL(dut))
+#define UART_NODE DT_NODELABEL(dut)
```

Comment:
Because of this: https://github.com/zephyrproject-rtos/zephyr/blob/a8178b21e54c78f1f4f4dd53d429014b4d7704c1/tests/drivers/uart/uart_async_api/src/test_uart.h#L22-L25
I wasn't sure if I could just add `dut` to `segger_rtt.overlay` and not break anything. Theoretically, this test could be run without this overlay but still with `CONFIG_UART_RTT` defined.

Reply:
Applied.